engine/runtime-benchmarks/src/main/java/org/enso/interpreter/bench/benchmarks/semantic/MultiValueBenchmarks.java

```diff
@@ -0,0 +1,133 @@
+package org.enso.interpreter.bench.benchmarks.semantic;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import org.enso.compiler.benchmarks.Utils;
+import org.graalvm.polyglot.Value;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.BenchmarkParams;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * These benchmarks compare performance of {@link EnsoMultiValue}. They create a vector in a certain
+ * configuration representing numbers and then they perform {@code sum} operation on it.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@Fork(1)
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+public class MultiValueBenchmarks {
+  private Value arrayOfNumbers;
+  private Value sum;
+  private Value self;
+  private final long length = 100000;
+
+  @Setup
+  public void initializeBenchmark(BenchmarkParams params) throws Exception {
+    var ctx = Utils.createDefaultContextBuilder().build();
+    var code =
+        """
+        from Standard.Base import Vector, Float, Number, Integer
+
+        type Complex
+            private Number re:Float im:Float
+
+        Complex.from (that:Number) = Complex.Number that 0
+
+        sum arr =
+            go acc i = if i >= arr.length then acc else
+                v = arr.at i : Float
+                sum = acc + v
+                @Tail_Call go sum i+1
+            go 0 0
+
+
+        make_vector type n =
```

Comment:
A dedicated benchmark for `EnsoMultiValue` instances based on the [idea of ArrayProxy one](https://github.com/enso-org/enso/issues/11846#issuecomment-2553634747). Currently the more complicated benchmarks _refuse to compile_ and the compiler bails out. The **initial results** are:
```
[info] Benchmark                                                 Mode  Cnt    Score    Error  Units
[info] MultiValueBenchmarks.sumOverComplexAndFloat5              avgt    5  214.330 ±  4.179  ms/op
[info] MultiValueBenchmarks.sumOverComplexFloatRecastedToFloat3  avgt    5  219.803 ± 11.872  ms/op
[info] MultiValueBenchmarks.sumOverFloat1                        avgt    5    0.079 ±  0.006  ms/op
[info] MultiValueBenchmarks.sumOverFloatAndComplex6              avgt    5  219.525 ±  6.393  ms/op
[info] MultiValueBenchmarks.sumOverFloatComplexRecastedToFloat4  avgt    5  203.788 ±  9.843  ms/op
[info] MultiValueBenchmarks.sumOverInteger0                      avgt    5    0.074 ±  0.001  ms/op
```

Reply:
Some [speedup achieved](https://github.com/enso-org/enso/pull/11924/files#r1894574140)...